src/RateLimiter/RateLimiter.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Web\RateLimiter;
+
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+/**
+ * RateLimiter limits the number of consequential requests ({@see RateLimiter::$limit}) that could be processed per
+ * {@see Counter::$interval}. If the number is exceeded, middleware responds with HTTP code 429, "Too Many Requests"
+ * until limit expires.
+ */
+final class RateLimiter implements MiddlewareInterface
+{
+    private int $limit = 1000;
+
+    private Counter $counter;
+
+    private ResponseFactoryInterface $responseFactory;
+
+    private bool $autoincrement = true;
```

Comment:
I don't really understand the point of leaving auto-increment. If autoincrement is disabled, it is more like just a "checker".